AK/SourceLocation.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2021, Andrew Kaster <andrewdkaster@gmail.com>
+ * Copyright (c) 2021, Brian Gianforcaro <bgianf@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/StringView.h>
+#include <AK/Types.h>
+
+namespace AK {
+
+class SourceLocation {
+public:
+    [[nodiscard]] constexpr StringView function_name() const { return StringView(m_function); }
+    [[nodiscard]] constexpr StringView file_name() const { return StringView(m_file); }
+    [[nodiscard]] constexpr u32 line_number() const { return m_line; }
+
+    [[nodiscard]] static constexpr SourceLocation current(const char* const file = __builtin_FILE(), u32 line = __builtin_LINE(), const char* const function = __builtin_FUNCTION())
+    {
+        return SourceLocation(file, line, function);
+    }
```

Comment:
I don't think having default arguments makes much sense here, otherwise you could make `current()` return something completely *not current*?

Suggestion:

```suggestion
    [[nodiscard]] static constexpr SourceLocation current()
    {
        return SourceLocation(__builtin_FILE(), __builtin_LINE(), __builtin_FUNCTION());
    }
```

for anything that should return a mix of current and custom values, you could still use the regular constructor with `__builtin_*()`.

Reply:
That won't work because file will always be `SourceLocation.h` and line will alwasy be `23`. `__builtin_FILE` / `__builtin_LINE` gives you the current source location except if you use it as default argument.

Reply:
oh duh, you're totally right :D